app/views/warehouse_reports/client_details/entries/_report.haml

```diff
@@ -25,37 +25,31 @@
       %table.mb-0.table
         %thead
           %tr
-            %th
-              Client
-            %th
-              Entry
-            %th
-              Exit
-            %th
-              Project Type
-            %th
-              Project
-            %th
-              Organization
+            %th Client
+            %th Entry
+            %th Exit
+            %th Project Type
+            %th Project
+            %th Organization
         %tbody
           - clients.each_with_index do |client_id, i|
             - meta = @report.enrollments[client_id].first
+            - client = meta.client
             - enrollments = @report.enrollments[client_id]
             - odd_class = if i % 2 == 0 then 'odd' else 'even' end
             %tr{class: odd_class}
               %td{rowspan: enrollments.count + 1}
-                = link_to_if current_user.can_view_clients?, "#{meta[:first_name]} #{meta[:last_name]}", appropriate_client_path(client_id)
+                = link_to_if current_user.can_view_clients?, "#{client.FirstName} #{client.LastName}", appropriate_client_path(client_id)
             - enrollments.each_with_index do |enrollment, index|
+              - project = enrollment.project
+              - organization = enrollment.organization
+              - safe_project_name = project.name(include_confidential_names: current_user.can_view_confidential_enrollment_details?)
```

Comment:
There is already a function `safe_project_name` on `Project`, could we simplify by making that function optionally check user permissions? That way we don't need to have the `can_view_confidential_enrollment_details` permission scattered all over the codebase, in case we change it.
Like:

```
safe_project_name  = project.safe_project_name(current_user)
```

Reply:
I debated that, didn't want to change the API because I _think_ that's in use elsewhere, but maybe I can just give it a default blank `user` and return the existing logic if that's empty.

Reply:
Oh, so in this case, the `safe_project_name` is being computed to determine if we can show the organization name.  I think I'll just make a `confidential_organization_name(user)` on project.  It would probably be cleaner to pass `user` to `project.name` but that's not always available, so we just pass in the permission (or sometimes true or false).